examples/multigpu_advtrain/target_attack.py

```diff
@@ -0,0 +1,257 @@
+                        adv_x = tf.stop_gradient(adv_x)
+                        outputs += [(x, adv_x)]
+
+        return inputs, outputs
+
+    def PGDcl(self, x, **kwargs):
+        eta = tf.truncated_normal(tf.shape(x), self.eps / 2)
+        eta = self.clip_eta(eta)
+        # Fix labels to the first model predictions for loss computation
+        model_preds = self.model.get_probs(x)
+        preds_max = tf.reduce_max(model_preds, 1, keep_dims=True)
+        if self.y is not None:
+            y = self.y
+        else:
+            y = tf.to_float(tf.equal(model_preds, preds_max))
+        y = tf.stop_gradient(y)
+
+        fgm_params = {'eps': self.eps_iter, 'y': y, 'ord': self.ord,
+                      'clip_min': self.clip_min, 'clip_max': self.clip_max}
+        for i in range(self.nb_iter):
+            FGM = FastGradientMethod(self.model, back=self.back,
+                                     sess=self.sess)
+            # Compute this step's perturbation
+            eta = FGM.generate(x + eta, **fgm_params) - x
+
+            eta = self.clip_eta(eta)
+
+        # Define adversarial example (and clip if necessary)
+        adv_x = x + eta
+        if self.clip_min is not None and self.clip_max is not None:
+            adv_x = tf.clip_by_value(adv_x, self.clip_min, self.clip_max)
+
+        return adv_x
+
+    def generate(self, x, **kwargs):
+        """
+        Generate symbolic graph for adversarial examples and return.
+        :param x: The model's symbolic inputs.
+        :param eps: (required float) maximum distortion of adversarial example
+                    compared to original input
+        :param eps_iter: (required float) step size for each attack iteration
+        :param nb_iter: (required int) Number of attack iterations.
+        :param y: (optional) A tensor with the model labels.
+        :param y_target: (optional) A tensor with the labels to target. Leave
+                         y_target=None if y is also set. Labels should be
+                         one-hot-encoded.
+        :param ord: (optional) Order of the norm (mimics Numpy).
+                    Possible values: np.inf, 1 or 2.
+        :param clip_min: (optional float) Minimum input component value
+        :param clip_max: (optional float) Maximum input component value
+        """
+
+        # Parse and save attack-specific parameters
+        assert self.parse_params(**kwargs)
+
+        # Initialize loop variables
+        if self.pgd:
+            if self.multigpu:
+                adv_x = self.PGDpgd_multigpu(x)
+            else:
+                adv_x = self.PGDpgd(x)
+        else:
+            adv_x = self.PGDcl(x)
+
+        return adv_x
+
+    def parse_params(self, eps=0.3, eps_iter=0.05, nb_iter=10, y=None,
+                     ord=np.inf, clip_min=None, clip_max=None,
+                     y_target=None, pgd=True, multigpu=False, ngpu=1,
+                     **kwargs):
+        """
```

Comment:
it looks like the docstring here is copy-pasted from another class and the new parameters like `pgd` aren't documented

Reply:
Fixed in 460d3f3.